src/data/matrix/notation.lean

```diff
@@ -231,4 +234,53 @@ by { ext i j, refine fin.cases _ _ i; simp [minor] }
 
 end minor
 
+section vec2_and_vec3
+
+variable [comm_ring α]  -- TODO get rid of
+
+lemma vec2_eq {a₀ a₁ b₀ b₁ : α} (h₀ : a₀ = b₀) (h₁ : a₁ = b₁) :
+  ![a₀, a₁] = ![b₀, b₁] :=
+by { ext x, fin_cases x; assumption }
+
+lemma vec3_eq {a₀ a₁ a₂ b₀ b₁ b₂ : α} (h₀ : a₀ = b₀) (h₁ : a₁ = b₁) (h₂ : a₂ = b₂) :
+  ![a₀, a₁, a₂] = ![b₀, b₁, b₂] :=
+by { ext x, fin_cases x; assumption }
+
+--variable [has_add α]
+
+lemma vec2_add {a₀ a₁ b₀ b₁ : α} :
+  ![a₀, a₁] + ![b₀, b₁] = ![a₀ + b₀, a₁ + b₁] :=
+by rw [cons_add_cons, cons_add_cons, empty_add_empty]
+
+lemma vec3_add {a₀ a₁ a₂ b₀ b₁ b₂ : α} :
+  ![a₀, a₁, a₂] + ![b₀, b₁, b₂] = ![a₀ + b₀, a₁ + b₁, a₂ + b₂] :=
+by rw [cons_add_cons, cons_add_cons, cons_add_cons, empty_add_empty]
+
+end vec2_and_vec3
+
+section vec2_and_vec3_dot
+
+localized "infix  ` • ` : 67 := matrix.dot_product" in matrix
```

Comment:
This is much worse than reusing the dot you had before, as this is `has_scalar.smul`!

Reply:
How about `⬝ᵥ`?

Reply:
Oh! I apologize; I didn't know about `has_scalar.smul`! I will fix it by changing ` • ` to ` ⬝ᵥ ` as you suggest.